controllers/actions.github.com/ephemeralrunner_controller.go

```diff
@@ -201,7 +202,32 @@ func (r *EphemeralRunnerReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 		default:
 			// Pod was not found. Create if the pod has never been created
 			log.Info("Creating new EphemeralRunner pod.")
-			return r.createPod(ctx, ephemeralRunner, secret, log)
+			result, err := r.createPod(ctx, ephemeralRunner, secret, log)
+			switch {
+			case err == nil:
+				return result, nil
+			case kerrors.IsInvalid(err) || kerrors.IsForbidden(err):
+				errMessage := fmt.Sprintf("Failed to create the pod: %v", err)
+				if err := r.markAsFailed(ctx, ephemeralRunner, errMessage, ReasonInvalidPodFailure, log); err != nil {
+					log.Error(err, "Failed to set ephemeral runner to phase Failed")
+					return ctrl.Result{}, err
+				}
+				return result, err
+			default:
+				errMessage := fmt.Sprintf("Failed to create the pod: %v", err)
+				log.Info("Failed to create the pod", "error", err)
+				if err := patchSubResource(ctx, r.Status(), ephemeralRunner, func(obj *v1alpha1.EphemeralRunner) {
+					if obj.Status.Failures == nil {
+						obj.Status.Failures = make(map[string]bool)
+					}
+					obj.Status.Failures[strconv.Itoa(len(ephemeralRunner.Status.Failures))] = true
+					obj.Status.Ready = false
+					obj.Status.Message = errMessage
+				}); err != nil {
+					log.Error(err, "failed to update ephemeral runner status: failed attempts")
+				}
+				return result, err
```

Comment:
This should not count as a failure. If there is a network issue, the controller should re-try again without keeping track of the failure. I would suggest just logging with error level and return an error. If the spec is not invalid, the kubernetes api will create a resource and wait for the kubelet. Once the actual pod fails, we should count that as failures. We should not count failed attempts to create a resource, unless the actual spec is invalid.
```suggestion
				log.Error(err, "Failed to create the pod")
				return ctrl.Result{}, err
```

Reply:
👍 Thanks for your explanation in detail.
6835393